base/download.jl

```diff
@@ -35,9 +35,19 @@ function find_curl()
     end
 end
 
+# Clear libpath for invoking system executables that we generally don't want to be confused
+# by things like a libcurl that sits in Julia's private libdir.
+function clear_libpath(cmd)
+    setenv(cmd,
+        "LD_LIBRARY_PATH" => nothing,
+        "DYLD_FALLBACK_LIBRARY_PATH" => nothing,
+        "DYLD_LIBRARY_PATH" => nothing,
+    )
```

Comment:
Don't you need something like

```suggestion
    env = copy(ENV)
    pop!(env, "LD_LIBRARY_PATH", nothing)
    pop!(env, "DYLD_FALLBACK_LIBRARY_PATH", nothing)
    pop!(env, "DYLD_LIBRARY_PATH", nothing)
    setenv(cmd, env)
```

?  I think the current implementation of `clear_libpath` can clear out some important environment variables (e.g., `PATH`). See:

```julia
julia> withenv("X" => "a") do
           cmd = `sh -c 'echo X = $X'`
           run(cmd)
           run(setenv(cmd, "Y" => "b"))
       end;
X = a
X =
```

It'd be nice to have API like `setenv(cmd, ENV, pairs...)` to do what we need here.